packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -210,14 +210,16 @@ export default class CashierStore extends BaseStore {
                 this.root_store.client.is_logged_in,
                 this.root_store.client.currency,
             ],
-            () => {
+            async () => {
+                // wait for get_settings so is_virtual gets populated in client-store
```

Comment:
`is_virtual` is from `authorize` response. Maybe we should wait for `authorize` here 🤔

Reply:
since we react to `this.root_store.client.is_logged_in` I think we should add  a check (`this.root_store.client.is_logged_in`) in line #217.
In this way we don't have to wait for `get_settings`

Reply:
We react to other two properties as well. In this case, the bug happens when we react to `this.root_store.client.switched` when a user is always logged in. In this case how adding `this.root_store.client.is_logged_in` would change the current logic? Also, we still need to keep checking for `is_virtual` as that's what logic is based on.

Reply:
Which `authorize` response do you refer to by the way?

Reply:
I think we should react to `this.root_store.client.login` instead of  `this.root_store.client.switched`

Reply:
Yes, correct. If we react to `loginId` we won't have to do this extra `get_settings` call

Reply:
@balakrishna-binary FYI, I had to revert this back to `switched` because now it seemed effecting the logic by reacting `too fast` before the currency value gets updated